### DISCUSSION TITLE
FBA or Merchant inventory

### DISCUSSION THREAD

**Discussion by matteoquintero** [amzn/selling-partner-api-models#322] [Q&A]
I am looking for a solution to determine if an Amazon seller has its inventory stored in FBA (Fulfillment by Amazon) or Merchant (sellers own shipping), without having to spend time consulting extensive reports. Is there an efficient way to get this information through the Amazon API? I would like to find the best way to query a seller's inventory without first knowing whether they use FBA or Merchant. I would greatly appreciate your advice and any recommendations on how to approach this challenge using the Amazon API.

**Comment by matteoquintero** [comment]
I have read the documentation well and the only thing I could do would be to consult different reports and from them draw a conclusion, but I think this solution may eventually fail as it is very subjective.

**Comment by ericblade** [comment]
As far as I'm aware, and IMO this is a major shortcoming that was totally missed in the API rebuild, the only way to get merchant fulfilled items is to pull a report like GET_MERCHANT_LISTINGS_DATA or one of it's similar friends, depending on which information you need about the items.

You can use /fbaInventory/getInventorySummaries to build a list of FBA info (there might also be a report for that, probably worth looking for, as it might more closely resemble GET_MERCHANT_LISTINGS_DATA ... but then again, it might not), but it's rate limit is pathetically low to use it for any real time data requests,
